### PR TITLE
Added no-undef rule to linter

### PR DESCRIPTION
**Pull request by not-an-aardvark**
It can be a bit annoying to keep track of the globals in the rc file, but I think it's worth it overall.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
   "parser": "babel-eslint",
   'env': {
-    'es6': true
+    'es6': true,
+    'node': true
```
Comment (YaManicKill) [inline .eslintrc]: It's not all node code (and you specify this in this file and in the client one)

Can you specify the client as browser and commonjs, and the api as node?

File: .eslintrc [hunk 2]
```diff
@@ -48,6 +49,8 @@
     'no-trailing-spaces': 2,
     // disallow declaration of variables that are not used in the code
     'no-unused-vars': [2, {'vars': 'local', 'args': 'none'}],
+    // require all variables to be defined
+    'no-undef': 2,
```
Comment (YaManicKill) [inline .eslintrc]: :+1:

**Comment by YaManicKill** [comment]
Looks good apart from the environment stuff, just fix that and I'm happy with it.

File: .eslintrc (hunk 1 above)
Comment (not-an-aardvark) [inline .eslintrc]: The issue is that there are a few other folders (config, test, etc.) which are node-only. I think it's better to leave `'node': true` as the default, and explicitly override it as necessary. I just pushed a commit to override it in the `client/` folder.

Comment (YaManicKill) [inline .eslintrc]: Yeah, that's fine. The main thing is that the client code isn't node :-)

**Pull request merged**